### PR TITLE
[receiver/kafkametrics] Fix metrics in kafkametricsreceiver (#4327)

### PR DESCRIPTION
**Pull request by ZenoCC-Peng**
**Description:** <Describe what has changed.> Utilize mdatagen to modify certain metrics from gauge to non-monotonic sum, employing delta aggregation. Adjust the test file to align with the sum test type and update the expected.yaml file for integration testing purposes.

**Link to tracking Issue:** <Issue number if applicable> [[4327](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/4327)]

**Testing:** <Describe what testing was performed and which tests were added.> Updated business logic and ran successfully`broker_scraper_test.go, config_test.go, consumer_scraper_test.go, factory_test.go, integration_test.go, receiver_test.go, topic_scraper_test.go`

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
I believe the following metrics need to be updated as well:
- kafka.topic.partitions
- kafka.partition.replicas
- kafka.partition.replicas_in_sync 
- kafka.consumer_group.members

**Comment by atoulme** [comment]
I have asked Zeno to start with just one, but once this lands, he can do all the others.

File: receiver/kafkametricsreceiver/metadata.yaml [hunk 1]
```diff
@@ -96,4 +98,4 @@ metrics:
     unit: 1
     gauge:
       value_type: int
-    attributes: [group, topic]
+    attributes: [group, topic]
```
Comment (dmitryax) [inline receiver/kafkametricsreceiver/metadata.yaml:101]: Nit: unnecessary removal of the training line. Usually, linters complain about not having them

**Comment by zeno-splunk** [comment]
> I believe the following metrics need to be updated as well:
> 
> * kafka.topic.partitions
> * kafka.partition.replicas
> * kafka.partition.replicas_in_sync
> * kafka.consumer_group.members



> I have asked Zeno to start with just one, but once this lands, he can do all the others.

Yes, I do have an alternative version for every other tag.

File: receiver/kafkametricsreceiver/metadata.yaml (hunk 1 above)
Comment (zeno-splunk) [reply receiver/kafkametricsreceiver/metadata.yaml:101]: Understand. 
This stems from an IDE, I think I should not enable autoformatting.

**Pull request merged**